### DISCUSSION TITLE
How would one go from converting exising yacc+lex / bison+flex grammars to .lang grammars?

### DISCUSSION THREAD

**Discussion by GitMensch** [jzimmerman/langcc#16] [Q&A] [answered]
Creating new language definitions from scratch can take quite some time, but as there are hundreds of parsers that use these other tools it would be quit useful to be able to "convert" them, at least for the start of the .lang file.

**Comment by jzimmerman** [comment, marked as answer]
In principle, one could define a `.lang` specification for yacc grammars, and write a compiler from yacc to `.lang`. However, I suspect this would not be very useful. In my experience, transcribing a grammar to `.lang` is not too time-consuming, and the process requires enough manual input (e.g., simplifying constructs to take advantage of langcc's features, rephrasing yacc recursion as langcc lists, deciding on the proper nested term structure (X.Y.Z)) that I don't think it would be feasible to usefully automate it.
